src/Components/TopContextBar.tsx

```diff
@@ -51,32 +53,36 @@ export const TopContextBar: React.FC<TopContextBarProps> = ({
             }
           : {})}
       >
-        {displayBackArrow && (
-          <ChevronIcon
-            direction="left"
-            height={14}
-            mr={1}
-            title=""
-            aria-hidden
-          />
-        )}
-
-        {image && (
-          <Box flexShrink={0} mr={1} borderRadius="50%" overflow="hidden">
-            <Image
-              src={image.src}
-              srcSet={image.srcSet}
-              width={30}
-              height={30}
-              alt=""
-              lazyLoad
+        <Flex flex={1} flexDirection="row" alignItems="center">
+          {displayBackArrow && (
+            <ChevronIcon
+              direction="left"
+              height={14}
+              mr={1}
+              title=""
+              aria-hidden
             />
-          </Box>
-        )}
+          )}
+
+          {image && (
+            <Box flexShrink={0} mr={1} borderRadius="50%" overflow="hidden">
+              <Image
+                src={image.src}
+                srcSet={image.srcSet}
+                width={30}
+                height={30}
+                alt=""
+                lazyLoad
+              />
+            </Box>
+          )}
+
+          <Text variant="xs" lineHeight={1} lineClamp={2}>
+            {children}
+          </Text>
+        </Flex>
 
-        <Text variant="xs" lineHeight={1} lineClamp={2}>
-          {children}
-        </Text>
+        {rightContent}
```

Comment:
Fine with this but just wondering — is the `rightContent` even necessary — could you just nest this layout inside the `children` as is?

Reply:
Yes, that's what I [originally did](https://github.com/artsy/force/pull/11150/commits/c34b2a1e72e2cedcfe34b27ee85decd49e75c0dc). but it seems to me that the current implementation will be better 😅

Reply:
It'd only be better if this was a pattern that we intend on using elsewhere. (Which is maybe the case? A question for the design team)